src/main/kotlin/graphics/scenery/utils/Image.kt

```diff
@@ -54,36 +72,134 @@ open class Image(val contents: ByteBuffer, val width: Int, val height: Int, val
             val pixels: IntArray
             val buffer: ByteArray
 
-            if (extension.lowercase().endsWith("tga")) {
-                try {
-                    val reader = BufferedInputStream(stream)
-                    buffer = ByteArray(stream.available())
-                    reader.read(buffer)
-                    reader.close()
-
-                    pixels = TGAReader.read(buffer, TGAReader.ARGB)
-                    val width = TGAReader.getWidth(buffer)
-                    val height = TGAReader.getHeight(buffer)
-                    bi = BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB)
-                    bi.setRGB(0, 0, width, height, pixels, 0, width)
-                } catch (e: IOException) {
-                    Colormap.logger.error("Could not read image from TGA. ${e.message}")
-                    bi = BufferedImage(1, 1, BufferedImage.TYPE_INT_ARGB)
-                    bi.setRGB(0, 0, 1, 1, intArrayOf(255, 0, 0), 0, 1)
+            when {
+
+                extension.lowercase().endsWith("tga") -> {
```

Comment:
The `extension.lowercase()` could be moved directly to the `when` as it's the same for each branch.